gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -3096,16 +3096,17 @@ nnstreamer_filter_shared_model_replace (void *instance, const char *key,
   GList *itr;
   UNUSED (instance);
 
-  if (!shared_model_table) {
-    ml_loge ("The shared model representation is not supported properly!");
-    return;
-  }
   if (!key) {
     ml_loge ("The key should NOT be NULL!");
     return;
   }
 
   G_LOCK (shared_model_table);
+  if (!shared_model_table) {
```

Comment:
Can you do G_LOCK if shared_model_table is NULL?

Reply:
Internally it is defined using glib macro G_LOCK_DEFINE_STATIC (shared_model_table).

Reply:
Oh ok... G_LOCK(name) does not use a variable named ```name```, but ```g__name__lock```.